modules/alloy/src/main/ts/ephox/alloy/api/ui/ModalDialog.ts

```diff
@@ -114,7 +116,13 @@ const factory: CompositeSketchFactory<ModalDialogDetail, ModalDialogSpec> = (det
         }),
         AddEventsBehaviour.config(modalEventsId, [
           AlloyEvents.runOnAttached((c) => {
-            AriaLabel.labelledBy(c.element, AlloyParts.getPartOrDie(c, detail, 'title').element);
+            // TINY-10808 - Workaround to address the dialog header not being announced on VoiceOver with aria-labelledby, ideally we should use the aria-labelledby
+            const title = TextContent.get(AlloyParts.getPartOrDie(c, detail, 'title').element);
+            if (browser.os.isMacOS() && Type.isNonNullable(title)) {
+              Attribute.set(c.element, 'aria-label', title);
+            } else {
+              AriaLabel.labelledBy(c.element, AlloyParts.getPartOrDie(c, detail, 'title').element);
```

Comment:
Is this macOS bug true of all `aria-labelledby` connections? If so you could move the fix into the `AriaLabel.labelledBy` function 🤔 

Also:
```suggestion
              AriaLabel.labelledBy(c.element, title);
```

Reply:
> Is this macOS bug true of all aria-labelledby connections

The other usage is in tabpanel, but they don't use this `AriaLabel`, only the dialog does.

The second parameter takes an element to set the id attribute instead of taking a string.